RIOT-OS/RIOT
# periph/dac: support for extension API

### Contribution description

This is an implementation of the extension API as proposed in #9582 for DAC extensions. It corresponds to the implementation of the GPIO extension API in #9860 and #9958. The PR contains the following contributions:

- Low level function prototypes `dac_*_ll` in `drivers/include/periph.h`.
- Changes of `dac_*` functions to redirect a function call either to the low level functions `dac_*_ll` or to the `dac_ext_*` function provided by an DAC extension device driver.
- New function prototype `dac_channels` which returns the number of channels, in case of low level drivers simply the `DAC_NUMOF`. This new function makes it possible to get the information from a certain device how many channels it has and allows to iterate over the all channels of all devices.
- Changes of low level function `dac_*_ll` for all CPUs that support DAC channels.

### Testing procedure

A test case is provided that implements a soft-driver for the DAC extension interface. The test case uses this driver to confirm that interception and redirection of the API call are working properly. This has been tested and is working properly on esp8266-esp-12x.

### Issues/PRs references

Implements #9582 
Corresponds to #9860 and #9958 

The changes in the `makefiles/pseudomodules.mk` and  in `drivers/Makefile.include` made to get it working are the same as in #9958. These changes might be removed once #9958 is merged.
